### ISSUE TITLE
optimize_acqf unexpected behavior

### ISSUE THREAD

**Issue by zhenlan0426** [pytorch/botorch#1845]
Trying to optimize some acq function using optimize_acqf. I notice that the best value returned is higher when optimizing over narrower bounds (strict subset). This is completely unexpected to me. Is that any reason this could happen. I try to increase num_restarts, does not change the results.

For context, I am trying to pre-apply a softmax to ensure the part of vector that correspond to a categorical inputs sums to one before passing them to GP. Then inputs for categorical GP would be over R^d ([-10,10] since optimize_acqf does not support inf). I notice that results are better when optimizing over [0,1].

x_best,y_val = optimize_acqf(acq,bounds,q=1,num_restarts=256,raw_samples=1024)

**Comment by Balandat** [comment]
Hmm as you said this shouldn't happen if the only thing you're changing are the `bounds` you're passing to `optimize_acqf`. 

What happens if you pass the location associated to the best value on the smaller domain as one of the [`batch_initial_conditions`](https://github.com/pytorch/botorch/blob/main/botorch/optim/optimize.py#L433)? Does this find the same maximizer?

**Comment by zhenlan0426** [comment]
Thank you for getting back. Bounds is the only difference between two optimize_acqf calls. I will close this issue because it would be hard for anyone else to diagnose without access to function I am optimizing. And it is for hyper-parameter tunning of ML model. So to replicate, I will have to share data.
